packages/react-components/src/components/VerticalGallery.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useTheme } from '../theming';
+import { BaseCustomStyles } from '../types';
+import {
+  childrenContainerStyle,
+  controlBarContainerStyle,
+  leftRightButtonStyles,
+  rootStyle
+} from './styles/VerticalGallery.styles';
+import { bucketize } from './utils/overFlowGalleriesUtils';
+
+export interface VerticalGalleryStyles extends BaseCustomStyles {
+  /** Styles for each video tile in the vertical gallery */
+  children?: IStyle;
+  /** Styles for the verticalGallery control bar */
+  controlBar?: VerticalGalleryControlBarStyles;
+}
+
+export interface VerticalGalleryControlBarStyles extends BaseCustomStyles {
+  /**
+   * styles for the next button in the VerticalGalleryControlBar
+   */
+  nextButton?: IStyle;
+  /**
+   * Styles for the previous button in the VerticalGalleryControlBar
+   */
+  previousButton?: IStyle;
+  /**
+   * Styles for the counter in the VerticalGalleryControlBar
+   */
+  counter?: IStyle;
+}
+/**
+ * Props for the VerticalGallery component
+ *
+ * @beta
+ */
+export interface VerticalGalleryProps {
+  /** Video tiles for the remote participants in the vertical gallery */
+  children: React.ReactNode;
+  /** Max number of children per page in the vertical Gallery */
+  childrenPerPage: number;
+  /** Styles to customize the vertical gallery */
+  styles?: VerticalGalleryStyles;
+}
+
+interface VerticalGalleryControlBarProps {
+  onNextButtonClick: () => void;
+  onPreviousButtonClick: () => void;
+  buttonsDisabled: { next: boolean; previous: boolean };
+  totalPages: number;
+  currentPage: number;
+  styles?: VerticalGalleryControlBarStyles;
+}
+
+/**
+ * VerticalGallery is a overflow gallery for participants in the {@link VideoGallery} component. Stacks
+ * participants on the Y-axis of the VideoGallery for better use of horizontal space.
+ *
+ * @beta
+ */
+export const VerticalGallery = (props: VerticalGalleryProps): JSX.Element => {
+  const { children, styles, childrenPerPage } = props;
+
+  const [page, setPage] = useState(0);
+  const [buttonState, setButtonState] = useState<{ previous: boolean; next: boolean }>({ previous: true, next: true });
+
+  const numberOfChildren = React.Children.count(children);
+  const lastPage = Math.ceil(numberOfChildren / childrenPerPage) - 1;
+
+  const paginatedChildren: React.ReactNode[][] = useMemo(() => {
+    return bucketize(React.Children.toArray(children), childrenPerPage);
+  }, [children, childrenPerPage]);
+
+  if (childrenPerPage <= 0) {
+    return <></>;
+  }
+
+  const firstIndexOfCurrentPage = page * childrenPerPage;
+  const clippedPage = firstIndexOfCurrentPage < numberOfChildren - 1 ? page : lastPage;
+  const childrenOnCurrentPage = paginatedChildren[clippedPage];
+
+  const showButtons = numberOfChildren > childrenPerPage;
+
+  const onPreviousButtonClick = () => {
+    setPage(page - 1);
+  };
+  const onNextButtonClick = () => {
+    setPage(page + 1);
+  };
+
+  useEffect(() => {
```

Comment:
Would this useEffect run when large amounts of people leave a call? 
Resulting in lower numberOfChildren?